mxfusion/components/functions/function_evaluation.py

```diff
@@ -120,13 +39,52 @@ def __init__(self, inputs, outputs, input_names, output_names,
             output_names=output_names)
 
     def replicate_self(self, attribute_map=None):
-        replicant = super(FunctionEvaluation, self).replicate_self(attribute_map)
+        replicant = super(
+            FunctionEvaluation, self).replicate_self(attribute_map)
         replicant.broadcastable = self.broadcastable
         return replicant
 
+    def eval(self, F, variables, always_return_tuple=False):
```

Comment:
Docstring? Explaining what this function does with samples in particular thanks.